### PR TITLE
Welcome stranger message can be improved in student home #1563

### LINKED ISSUE #1563: Welcome stranger message can be improved in student home
_From [wulifu20...@gmail.com](https://code.google.com/u/107379343006199273595/) on January 27, 2014 23:37:05_

What can be improved on the system? What can be added to the system? 1. The welcome stranger message doesn't attract the attention of the users very well. It doesn't make me feel that something is going wrong and I better take a look at it
2. the welcome message in the student home page can be improved to be more readable (a lot of information, no highlight) Please specify: 1) Exact page to implement/improve feature after student login (student home) 2) The details of the feature a) show the "welcome stranger" msg in a way that attracts people's attention. eg. show a big notice such as "oops, you are not a registered student"
b) highlight the important sentence (eg. make it bold or use another color)

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1564_

### PR DESCRIPTION
**Pull request by thyageshm**
Fixes #1563

image

### CODE REVIEW

**Comment by thyageshm** [comment]
image
I am getting this unrelated failure in browsertests

**Comment by wkurniawan07** [comment]
@thyageshm that looks like your dev server is not in UTC

**Comment by thyageshm** [comment]
Hahahaha I just realised... and edited my post

File: src/main/java/teammates/common/util/Const.java [hunk 1]
```diff
@@ -1138,24 +1138,31 @@ public static final String getjQueryUiFilePath(boolean isDevServer) {
...
         
-        public static final String STUDENT_FIRST_TIME = "<div style=\"text-align:left;\">Welcome stranger :-) "
-                + "<br><br>It seems you are not a registered user of TEAMMATES. "
-                + "To use TEAMMATES, a course instructor has to add you to a course first. "
-                + "After that, TEAMMATES will send you an email containing the link to 'join' that course. "
-                + "<br><br>Not a stranger to TEAMMATES? "
-                + "Could log in before, but not any more? These are the possible reasons:"
-                + "<br>1. You used a different Google account to access TEAMMATES in the past. "
+        public static final String STUDENT_FIRST_TIME = "<div class='align-left'>"
+                + "<div class='align-center text-color-red text-bold '>"
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/Const.java]: Three things:
- Use double-quote for HTML attributes, as how it was done before. Also applies to the line before.
- There's an extra space after `text-bold`.
- To improve the readability of this section, format it like this (e.g. preserve the HTML indentations):
```java
public static final String STUDENT_FIRST_TIME =
          "<div>"
            + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor "
            + "<div>"
                + "incididunt ut labore et dolore magna aliqua."
            + "</div>"
        + "</div>";
```

**Comment by thyageshm** [comment]
@wkurniawan07 ready for review

File: src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java [hunk 2]
```diff
@@ -214,7 +214,8 @@ private void verifyRedirectToWelcomeStrangerPage(AppUrl url, String unregUsernam
         currentPage.navigateTo(url);
         // A simple regex check is enough because we do full HTML tests
         // elsewhere
-        AssertHelper.assertContainsRegex("{*}" + unregUsername + "{*}Welcome stranger{*}",
+        AssertHelper.assertContainsRegex("{*}" + unregUsername
+                + "{*}Ooops! Your Google account is not known to TEAMMATES{*}",
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: For this and also the changes happening in `StudentHomePageActionTest.java`, this kind of line break looks better:
```java
AssertHelper.assertContainsRegex(
        "{*}" + unregUsername + "{*}Ooops! Your Google account is not known to TEAMMATES{*}",
        currentPage.getPageSource());
```

File: src/main/java/teammates/common/util/Const.java [hunk 3]
```diff
@@ -1138,24 +1138,38 @@ public static final String getjQueryUiFilePath(boolean isDevServer) {
...
+                            + "If you don't remember which Google account you used previously, "
+                            + "email us from the same email account to which you receive TEAMMATES emails."
+                        + "</li>"
+                        + "<li>"
+                            + "You changed the primary email from a non-Gmail address to a Gmail address recently. "
+                            + "In that case, <a href=\"/contact.jsp\">email us</a> "
+                            + "so that we can reconfigure your account to use the new Gmail address."
+                        + "</li>"
+                
+                        + "<li>You joined this course just a few seconds ago and your data "
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/Const.java]: This part is not standardized in two manners:
- There is an empty line in between `</li>...<li>`, not present previously. (either have an empty line for both, or for none)
- The content after `<li>` here is not line-broken as how it is done for the previous two list items.

File: src/main/java/teammates/common/util/Const.java [hunk 4]
```diff
@@ -1138,24 +1138,38 @@ public static final String getjQueryUiFilePath(boolean isDevServer) {
...
-                + "may be still in the process of propagating through our servers. "
-                + "In that case, please click on the "
-                + "<a href=" + ActionURIs.STUDENT_HOME_PAGE + ">Home</a> link above in a few minutes. "
+        public static final String STUDENT_FIRST_TIME =
+                "<div class=\"align-left\">"
+                    + "<div class=\"align-center text-color-red text-bold\">"
+                        + "Ooops! Your Google account is not known to TEAMMATES"
+                    + "</div>"
+                    + "<br>"
+                    + "To access a course on TEAMMATES, you need to wait till an instructor adds you to "
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/Const.java]: "till" sounds informal. "until" should be better? You don't need to follow if you disagree.

File: src/test/java/teammates/test/cases/ui/StudentHomePageActionTest.java [hunk 5]
```diff
@@ -84,8 +85,9 @@ public void testExecuteAndPostProcess() throws Exception {
                                           + studentWithoutCourses.googleId,
                                           r.getDestinationWithParams());
         assertFalse(r.isError);
-        AssertHelper.assertContainsRegex("Welcome stranger :-){*}use the new Gmail address.",
-                                          r.getStatusMessage());
+        AssertHelper.assertContainsRegex(
+                "Ooops! Your Google account is not known to TEAMMATES" + "{*}use the new Gmail address.",
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/StudentHomePageActionTest.java]: Here it doesn't need to be separated, just `...TEAMMATES{*}use...`. Similarly for few lines above.

File: src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java [hunk 6]
```diff
@@ -214,7 +214,8 @@ private void verifyRedirectToWelcomeStrangerPage(AppUrl url, String unregUsernam
         currentPage.navigateTo(url);
         // A simple regex check is enough because we do full HTML tests
         // elsewhere
-        AssertHelper.assertContainsRegex("{*}" + unregUsername + "{*}Welcome stranger{*}",
+        AssertHelper.assertContainsRegex(
+                "{*}" + unregUsername + "{*}Ooops! Your Google account is not known to TEAMMATES{*}",
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: Actually, even it was done this way before, the leading and trailing `{*}` should not be necessary, right?

File: src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java [hunk 7]
```diff
@@ -214,7 +214,8 @@ private void verifyRedirectToWelcomeStrangerPage(AppUrl url, String unregUsernam
         currentPage.navigateTo(url);
         // A simple regex check is enough because we do full HTML tests
         // elsewhere
-        AssertHelper.assertContainsRegex("{*}" + unregUsername + "{*}Welcome stranger{*}",
+        AssertHelper.assertContainsRegex(
+                unregUsername + "{*}Ooops! Your Google account is not known to TEAMMATE",
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: Minor issue but you accidentally remove TEAMMATE~~S~~ as well :p

**Comment by wkurniawan07** [comment]
@thyageshm no other comments from me, you can set it to `s.ToMerge` after correcting the minor mistake.

File: src/test/java/teammates/test/cases/ui/StudentHomePageActionTest.java [hunk 8]
```diff
@@ -46,8 +46,9 @@ public void testExecuteAndPostProcess() throws Exception {
         AssertHelper.assertContainsRegex("/jsp/studentHome.jsp?error=false&user=unreg.user",
                                         r.getDestinationWithParams());
         assertFalse(r.isError);
-        AssertHelper.assertContainsRegex("Welcome stranger :-){*}use the new Gmail address.",
-                                        r.getStatusMessage());
+        AssertHelper.assertContainsRegex(
+                "Ooops! Your Google account is not known to TEAMMATES{*}use the new Gmail address.",
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/StudentHomePageActionTest.java]: Does the word `new` make sense here?

**Comment by damithc** [comment]
Waiting for a clarification

**Comment by thyageshm** [comment]
@damithc so can the pr be merged?

**Comment by damithc** [comment]
Yes, go ahead.

**Pull request merged**